homeassistant/components/apcupsd/__init__.py

```diff
@@ -99,6 +90,22 @@ def statflag(self) -> str | None:
         """Return the STATFLAG indicating the status of the UPS, if available."""
         return self.status.get("STATFLAG")
 
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return the DeviceInfo of this APC UPS for the sensors, if serial number is available."""
+        if self.serial_no is None:
+            return None
```

Comment:
For some really dumb APC UPSes, the APCUPS daemon does not report the serial number, but some other basic information (such as UPS status) is available (see http://www.apcupsd.org/manual/)

> For a simple signaling or dumb UPS such as BackUPS, your output will be very minimal as follows:
> 
> APC      : 001,012,0319
> DATE     : Mon Feb 18 09:11:50 CST 2002
> RELEASE  : 3.8.5
> UPSNAME  : UPS_IDEN
> CABLE    : APC Cable 940-0128A
> MODEL    : BackUPS
> UPSMODE  : Stand Alone
> STARTTIME: Mon Feb 18 09:11:45 CST 2002
> LINEFAIL : OK
> BATTSTAT : OK
> STATFLAG : 0x008 Status Flag
> END APC  : Mon Feb 18 09:15:01 CST 2002

This minimal APCUPSD response is also included as a test case.

Reply:
Is there another way we can get an identifier for this UPS? (Maybe even the config entry id would be good enough for this). Otherwise you have a lot of entities without device.